openpype/hosts/maya/api/pipeline.py

```diff
@@ -40,49 +50,121 @@
+        register_event_callback("open", on_open)
+        register_event_callback("new", on_new)
+        register_event_callback("before.save", on_before_save)
+        register_event_callback("taskChanged", on_task_changed)
+        register_event_callback("workfile.save.before", before_workfile_save)
+
+    def open_workfile(self, filepath):
+        return open_file(filepath)
+
+    def save_workfile(self, filepath=None):
+        return save_file(filepath)
 
-    _set_project()
-    _register_callbacks()
+    def work_root(self, session):
+        return work_root(session)
 
-    menu.install()
+    def get_current_workfile(self):
+        return current_file()
 
-    register_event_callback("save", on_save)
-    register_event_callback("open", on_open)
-    register_event_callback("new", on_new)
-    register_event_callback("before.save", on_before_save)
-    register_event_callback("taskChanged", on_task_changed)
-    register_event_callback("workfile.save.before", before_workfile_save)
+    def workfile_has_unsaved_changes(self):
+        return has_unsaved_changes()
+
+    def get_workfile_extensions(self):
+        return file_extensions()
+
+    def get_containers(self):
+        return ls()
+
+    @contextlib.contextmanager
+    def maintained_selection(self):
+        with lib.maintained_selection():
+            yield
+
+    def _register_callbacks(self):
+        for handler, event in self._op_events.copy().items():
+            if event is None:
+                continue
+
+            try:
+                OpenMaya.MMessage.removeCallback(event)
+                self._op_events[handler] = None
+            except RuntimeError as exc:
+                self.log.info(exc)
+
+        self._op_events[_on_scene_save] = OpenMaya.MSceneMessage.addCallback(
+            OpenMaya.MSceneMessage.kBeforeSave, _on_scene_save
+        )
+
+        self._op_events[_before_scene_save] = (
+            OpenMaya.MSceneMessage.addCheckCallback(
+                OpenMaya.MSceneMessage.kBeforeSaveCheck,
+                _before_scene_save
+            )
+        )
+
+        self._op_events[_on_scene_new] = OpenMaya.MSceneMessage.addCallback(
+            OpenMaya.MSceneMessage.kAfterNew, _on_scene_new
+        )
+
+        self._op_events[_on_maya_initialized] = (
+            OpenMaya.MSceneMessage.addCallback(
+                OpenMaya.MSceneMessage.kMayaInitialized,
+                _on_maya_initialized
+            )
+        )
+
+        self._op_events[_on_scene_open] = OpenMaya.MSceneMessage.addCallback(
+            OpenMaya.MSceneMessage.kAfterOpen, _on_scene_open
+        )
+
+        self.log.info("Installed event handler _on_scene_save..")
+        self.log.info("Installed event handler _before_scene_save..")
+        self.log.info("Installed event handler _on_scene_new..")
+        self.log.info("Installed event handler _on_maya_initialized..")
+        self.log.info("Installed event handler _on_scene_open..")
```

Comment:
not sure about the log leve here, nor the text. The is mainly interesting for developer so it should be debug. Or if not, it should be more user friendly than `_on_scene_save..` that will remind artist of an ascii art.

Reply:
All of them were there before this PR. Irrelevant from point of this PR.

Reply:
I agree. As much as there refactoring PRs uncover a lot of tech debt, it should be logged into Issues and handled separately. To keep focus on what these PRs are doing.